ui/packages/atlasmap/src/Atlasmap/dialogs/useImportDocumentDialog.tsx

```diff
@@ -2,25 +2,76 @@ import { useCallback, ReactElement } from "react";
+                          {label: 'MongoDBCsv', value: 'MongoDBCsv'}, {label: 'MongoDBTsv', value: 'MongoDBTsv'},
+                          {label: 'MySQL', value: 'MySQL'}, {label: 'Oracle', value: 'Oracle'},
+                          {label: 'PostgreSQLCsv', value: 'PostgreSQLCsv'},
+                          {label: 'PostgreSQLText', value: 'PostgreSQLText'},
+                          {label: 'RFC4180', value: 'RFC4180'},
+                          {label: 'TDF', value: 'TDF'}], hideable: false},
+                  {name: 'allowDuplicateHeaderNames', value: 'true', options: [{label: 'true', value: 'true'},
+                          {label: 'false', value: 'false'}], hidden: true, hideable: true},
+                  {name: 'allowMissingColumnNames', value: 'true', options: [{label: 'true', value: 'true'},
+                          {label: 'false', value: 'false'}], hidden: true, hideable: true},
+                  {name: 'commentMarker', value: '', hidden: true, hideable: true},
+                  {name: 'delimiter', value: '', hidden: true, hideable: true},
+                  {name: 'escape', value: '', hidden: true, hideable: true},
+                  {name: 'firstRecordAsHeader', value: 'true', options: [{label: 'true', value: 'true'},
+                          {label: 'false', value: 'false'}], hidden: true, hideable: true},
+                  {name: 'headers', value: '', hidden: true, hideable: true},
+                  {name: 'ignoreEmptyLines', value: 'true', options: [{label: 'true', value: 'true'},
+                          {label: 'false', value: 'false'}], hidden: true, hideable: true},
+                  {name: 'ignoreHeaderCase', value: 'true', options: [{label: 'true', value: 'true'},
+                          {label: 'false', value: 'false'}], hidden: true, hideable: true},
+                  {name: 'ignoreSurroundingSpaces', value: 'true', options: [{label: 'true', value: 'true'},
+                          {label: 'false', value: 'false'}], hidden: true, hideable: true},
+                  {name: 'nullString', value: '', hidden: true, hideable: true},
+                  {name: 'quote', value: '', hidden: true, hideable: true}
+                  ]);
+              return;
+          }
+      }
+
+      importAtlasFile(selectedFile, isSource);
+  };
+
   const onImportDocument = useCallback(
     (selectedFile: File, isSource: boolean) => {
       if (documentExists(selectedFile, isSource)) {
-        openImportDialog(() => importAtlasFile(selectedFile, isSource));
+        openImportDialog(() => importFile(selectedFile, isSource));
       } else {
-        importAtlasFile(selectedFile, isSource);
+        importFile(selectedFile, isSource);
       }
     },
-    [documentExists, importAtlasFile, openImportDialog],
+    [documentExists, importAtlasFile, openImportDialog, openParametersDialog],
   );
-  return [importDialog, onImportDocument];
+  return [importDialog, parametersDialog, onImportDocument];
```

Comment:
I think we can avoid changing the return type by wrapping the two dialogs together in a fragment:

```tsx
return [<>{importDialog}{parametersDialog}</>, onImportDocument];
```

Having two dialogs is an implementation detail and it would be nice not to leak it outside this function.